### DISCUSSION TITLE
React InstantSearch Hooks experimental is available for wider use

### DISCUSSION THREAD

**Discussion by francoischalifour** [algolia/react-instantsearch#3267] [Announcements]
We're super excited to announce that **[React InstantSearch Hooks](https://algolia.com/doc/guides/building-search-ui/what-is-instantsearch/react-hooks/) is now available for wider use**.

React InstantSearch Hooks provides the power of [InstantSearch.js](https://algolia.com/doc/guides/building-search-ui/what-is-instantsearch/js/) with a thin React compatibility layer. You get to use [React Hooks](https://reactjs.org/docs/hooks-intro.html) to power your search experience.

You can find more on the Algolia documentation:

- [What is React InstantSearch Hooks?](https://algolia.com/doc/guides/building-search-ui/what-is-instantsearch/react-hooks/)
- [Installation](https://algolia.com/doc/guides/building-search-ui/installation/react-hooks/)
- [Server-side rendering](https://algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react-hooks/)
- [API reference](https://algolia.com/doc/api-reference/widgets/react-hooks/)

Check out the [live example](https://codesandbox.io/s/github/algolia/react-instantsearch/tree/master/examples/hooks) to play with the library.

Consider using React InstantSearch Hooks **if your app isn’t meant to be production-ready** yet, and:

- you have an existing React InstantSearch application that mainly relies on InstantSearch connectors,
- or you’re creating a new InstantSearch application with React.

Eventually, React InstantSearch Hooks is planned to become the main React implementation of InstantSearch.js.

Keep us updated with the awesome search experiences you build!

**Comment by wenche** [comment]
Hello!

We are building a new application  from scratch and we decided to try out the hooks version of the library this time. But right now I'm just a little bit confused about what to expect from the final release. 😅 

Are you able to share something about what I can expect from the production ready version of this library? Our impression was that it was a pure hooks library _without_ UI components. But when I look at the [React Instantsearch repository](https://github.com/algolia/react-instantsearch/tree/master/packages) I notice that you are working on several packages in parallel, namely `react-instantsearch-hooks`, `react-instantsearch-hooks-server` and `react-instantsearch-hooks-dom `, the latter not public available yet. 

My question is: It seems like you are creating components as well (like `Pagination` today). Are these components intended to be used as UI building blocks in the same way as in `react-instantsearch-dom` in existing library? If yes, it seems like we should just pause the search implementation for a little while, but for how long? Do you have a timeline/roadmap for when it will be production ready? 🤔 

Lot's of nice features with the hooks btw 👍

**Reply by Haroenv** [reply]
Thanks for the question. We are indeed working on the UI components, but as we are reviewing the implementation of each one of them, it will likely take a few months to get to a fully final version with all components available in the existing React InstantSearch DOM package.

I'd recommend for now to use our example as a guide for your own components, and then switch to hooks dom once that's released. Almost every component is available as an example here: https://github.com/algolia/react-instantsearch/tree/master/examples/hooks/components

Please note that the React InstantSearch Hooks package will stay marked as experimental until we have finished the implementation of all dom components and have had feedback from larger implementations that everything works as expected.

If you have any feedback, feel free to open issues, and we will be able to address those before releasing as a GA.

Thanks for trying out React InstantSearch Hooks @wenche!
